haystack/components/preprocessors/utils.py

```diff
@@ -0,0 +1,233 @@
+        """
+        Applies additional split rules to the sentence spans.
+
+        :param text: The text to split.
+        :param sentence_spans: The list of sentence spans to split.
+        :returns: The list of sentence spans after applying the split rules.
+        """
+        new_sentence_spans = []
+        quote_spans = [match.span() for match in QUOTE_SPANS_RE.finditer(text)]
+        while sentence_spans:
+            span = sentence_spans.pop(0)
+            next_span = sentence_spans[0] if len(sentence_spans) > 0 else None
+            while next_span and SentenceSplitter._needs_join(text, span, next_span, quote_spans):
+                sentence_spans.pop(0)
+                span = (span[0], next_span[1])
+                next_span = sentence_spans[0] if len(sentence_spans) > 0 else None
+            start, end = span
+            new_sentence_spans.append((start, end))
+        return new_sentence_spans
+
+    @staticmethod
+    def _needs_join(
+        text: str, span: Tuple[int, int], next_span: Tuple[int, int], quote_spans: List[Tuple[int, int]]
+    ) -> bool:
+        """
+        Checks if the spans need to be joined as parts of one sentence.
+
+        :param text: The text containing the spans.
+        :param span: The current sentence span within text.
+        :param next_span: The next sentence span within text.
+        :param quote_spans: All quoted spans within text.
+        :returns: True if the spans needs to be joined.
+        """
+        start, end = span
+        next_start, next_end = next_span
+
+        # sentence. sentence"\nsentence -> no split (end << quote_end)
+        # sentence.", sentence -> no split (end < quote_end)
+        # sentence?", sentence -> no split (end < quote_end)
+        if any(quote_start < end < quote_end for quote_start, quote_end in quote_spans):
+            # sentence boundary is inside a quote
+            return True
+
+        # sentence." sentence -> split (end == quote_end)
+        # sentence?" sentence -> no split (end == quote_end)
+        if any(quote_start < end == quote_end and text[quote_end - 2] == "?" for quote_start, quote_end in quote_spans):
+            # question is cited
+            return True
+
+        if re.search(r"(^|\n)\s*\d{1,2}\.$", text[start:end]) is not None:
+            # sentence ends with a numeration
+            return True
+
+        # next sentence starts with a bracket or we return False
+        return re.search(r"^\s*[\(\[]", text[next_start:next_end]) is not None
+
+    @staticmethod
+    def _read_abbreviations(language: Language) -> List[str]:
+        """
+        Reads the abbreviations for a given language from the abbreviations file.
+
+        :param language: The language to read the abbreviations for.
+        :returns: List of abbreviations.
+        """
+        abbreviations_file = Path(__file__).parent.parent / f"data/abbreviations/{language}.txt"
```

Comment:
Hey @vblagoje I noticed that we didn't add these files, could we do that?

Reply:
You can find them here https://github.com/deepset-ai/deepset-cloud-custom-nodes/tree/haystack-1.x/deepset_cloud_custom_nodes/data/abbreviations